datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/mutate/MutationUtils.java

```diff
@@ -99,37 +101,63 @@ public static EditableSchemaFieldInfo getFieldInfoFromSchema(
     }
   }
 
-  public static Boolean validateSubresourceExists(
+  public static Optional<SchemaField> getSchemaFieldFromSubResource(
       Urn targetUrn,
       String subResource,
       SubResourceType subResourceType,
       EntityService entityService
   ) {
-    if (subResourceType.equals(SubResourceType.DATASET_FIELD)) {
-      SchemaMetadata schemaMetadata = (SchemaMetadata) entityService.getAspect(targetUrn, SCHEMA_ASPECT_NAME, 0);
+    if (subResourceType.equals(SubResourceType.DATASET_FIELD) || subResourceType.equals(SubResourceType.FIELD)) {
 
-      if (schemaMetadata == null) {
-        throw new IllegalArgumentException(
-            String.format("Failed to update %s & field %s. %s has no schema.", targetUrn, subResource, targetUrn)
-        );
-      }
+      if (targetUrn.getEntityType().equals(Constants.DATASET_ENTITY_NAME)) {
+        SchemaMetadata schemaMetadata = (SchemaMetadata) entityService.getAspect(targetUrn, SCHEMA_ASPECT_NAME, 0);
 
-      Optional<SchemaField> fieldMatch =
-          schemaMetadata.getFields().stream().filter(field -> field.getFieldPath().equals(subResource)).findFirst();
+        if (schemaMetadata == null) {
+          throw new IllegalArgumentException(String.format("Failed to update %s & field %s. %s has no schema.", targetUrn, subResource, targetUrn));
+        }
 
-      if (!fieldMatch.isPresent()) {
-        throw new IllegalArgumentException(String.format(
-            "Failed to update %s & field %s. Field %s does not exist in the datasets schema.",
-            targetUrn, subResource, subResource));
+        Optional<SchemaField> fieldMatch =
+            schemaMetadata.getFields().stream().filter(field -> field.getFieldPath().equals(subResource)).findFirst();
+        return fieldMatch;
       }
 
-      return true;
```

Comment:
this diff seems hard to read but it was essentially just factoring some logic out of `validateSubresourceExists` and into a helper method `getSchemaFieldFromSubResource`